crates/sui-types/src/nitro_attestation.rs

```diff
@@ -0,0 +1,622 @@
+    }
+
+    /// Verify the certificate against AWS Nitro root of trust and checks expiry.
+    fn validate_cert(&self, now: u64) -> Result<(), NitroError> {
+        // Create chain starting with leaf cert all the way to root.
+        let mut chain = Vec::with_capacity(1 + self.cabundle.len());
+        chain.push(self.certificate.as_slice());
+        chain.extend(self.cabundle.iter().rev().map(|cert| cert.as_slice()));
+        validate_cert_chain(&chain, now)
+    }
+
+    /// Validate the PCRs against the expected PCRs.
+    fn validate_pcrs(&self, expected_pcrs: &[&[u8]]) -> Result<(), NitroError> {
+        if expected_pcrs.is_empty() || expected_pcrs.len() > 32 {
+            return Err(NitroError::InvalidPcrs);
+        }
+        for (i, expected_pcr) in expected_pcrs.iter().enumerate() {
+            if self.pcrs[i] != *expected_pcr {
+                return Err(NitroError::InvalidPcrs);
+            }
+        }
+        Ok(())
+    }
+}
+
+/// Validate the certificate chain against the root of trust.
+fn validate_cert_chain(cert_chain: &[&[u8]], now_ms: u64) -> Result<(), NitroError> {
+    if cert_chain.is_empty() || cert_chain.len() > 20 {
+        return Err(NitroError::InvalidCertificate(
+            "No certificate chain".to_string(),
+        ));
+    }
+
+    let root_cert = X509Certificate::from_der(ROOT_CERTIFICATE.as_slice())
+        .map_err(|e| NitroError::InvalidCertificate(e.to_string()))?
+        .1;
+    let now_secs = now_ms / 1000;
+
+    // Validate the chain starting from the leaf
+    for i in 0..cert_chain.len() {
+        let cert = X509Certificate::from_der(cert_chain[i])
+            .map_err(|e| NitroError::InvalidCertificate(e.to_string()))?
+            .1;
+
+        // Check timestamp in seconds.
+        if now_secs < cert.validity().not_before.timestamp() as u64 {
+            return Err(NitroError::InvalidCertificate(
+                "Certificate not yet valid".to_string(),
+            ));
+        }
+        if now_secs > cert.validity().not_after.timestamp() as u64 {
+            return Err(NitroError::InvalidCertificate(
+                "Certificate expired".to_string(),
+            ));
+        }
+
+        // Get issuer cert from either next in chain or root
+        let issuer_cert = if i < cert_chain.len() - 1 {
+            X509Certificate::from_der(cert_chain[i + 1])
+                .map_err(|e| NitroError::InvalidCertificate(e.to_string()))?
+                .1
+        } else {
+            root_cert.clone()
+        };
+
+        // Verify issuer/subject chaining.
+        if cert.issuer() != issuer_cert.subject() {
+            return Err(NitroError::InvalidCertificate(
+                "Certificate chain issuer mismatch".to_string(),
+            ));
+        }
+
+        // Verify certificate signature.
+        let verifying_key = match issuer_cert.public_key().parsed() {
+            Ok(PublicKey::EC(ec)) => VerifyingKey::from_sec1_bytes(ec.data()).map_err(|_| {
```

Comment:
The previous implementation worked with any signature scheme for the cert chain, no? is it safe to assume that future cert chains will use p384 as well? i think it's ok, but let's document the right reference for that.